### PR TITLE
Test cases added for Bitbucket Repository Service

### PR DESCRIPTION
**Pull request by maksudur-rahman-maruf**
1. Field name changed in Automation class.

2. Test cases added for BitbucketRepositoryService class.

3. Refactored BitbucketRepositoryService class.

### CODE REVIEW

File: src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java [hunk 1]
```diff
@@ -84,4 +70,27 @@ public Bitbucket fetchBitbucketRepositoryDetails(Jenkins jenkins, Map<String, Ob
         return bitbucket;
     }
 
+    public static Bitbucket createBitbucketObject(String serverUrl, String bitBucketToken, Integer projectRepositoryPullNumber, String repositoryName, String projectKey) {
+        Bitbucket bitbucket = new Bitbucket();
+        Api bitbucketApi = new Api();
```
Comment (zaman-akib) [inline src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java:75]: Do we need to create an `Api` object again?  I see it's already instantiated with `Bitbucket` constructor.

File: src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java [hunk 2]
```diff
@@ -84,4 +70,27 @@ public Bitbucket fetchBitbucketRepositoryDetails(Jenkins jenkins, Map<String, Ob
...
+        bitbucketApi.setUrl(serverUrl);
+        bitbucketApi.setToken(bitBucketToken);
```
Comment (zaman-akib) [inline src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java:77]: We can use these like - ` bitbucket.getApi().setUrl(serverUrl)`

File: src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java [hunk 3]
```diff
@@ -84,4 +70,27 @@ public Bitbucket fetchBitbucketRepositoryDetails(Jenkins jenkins, Map<String, Ob
...
+
+        Pull pull = new Pull();
+        pull.setNumber(projectRepositoryPullNumber);
+
+        Repository repository = new Repository();
+        repository.setName(repositoryName);
+        repository.setPull(pull);
+
+        Project project = new Project();
+        project.setKey(projectKey);
```
Comment (zaman-akib) [inline src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java:87]: Similar as above mentioned - ` bitbucket.getProject().setKey(projectKey)`

File: src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java [hunk 4]
```diff
@@ -84,4 +70,27 @@ public Bitbucket fetchBitbucketRepositoryDetails(Jenkins jenkins, Map<String, Ob
...
+        project.setRepository(repository);
+
+        bitbucket.setApi(bitbucketApi);
+        bitbucket.setProject(project);
```
Comment (zaman-akib) [inline src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java:91]: Then we don't need these

**Review by zaman-akib — CHANGES_REQUESTED** [review]
We may need to check if we need Tests for `SCMRepositoryService` class.

But first this PR needs to be merged as we need these changes for prcomment testing. Other tests can be in another PR.

File: src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java (hunk 2 above)
Comment (maksudur-rahman-maruf) [reply src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java:77]: Thank you for the suggestion.

**Pull request merged**